source/slang/slang-emit-metal.cpp

```diff
@@ -298,24 +298,26 @@ bool MetalSourceEmitter::tryEmitInstExprImpl(IRInst* inst, const EmitOpInfo& inO
 
 void MetalSourceEmitter::emitVectorTypeNameImpl(IRType* elementType, IRIntegerValue elementCount)
 {
-    switch (elementType->getOp())
-    {
-    case kIROp_FloatType:  m_writer->emit("float");  break;
-    case kIROp_HalfType:   m_writer->emit("half");   break;
-    case kIROp_BoolType:   m_writer->emit("bool");   break;
-    case kIROp_Int8Type:   m_writer->emit("char");   break;
-    case kIROp_UInt8Type:  m_writer->emit("uchar");  break;
-    case kIROp_Int16Type:  m_writer->emit("short");  break;
-    case kIROp_UInt16Type: m_writer->emit("ushort"); break;
-    case kIROp_IntType:    m_writer->emit("int");    break;
-    case kIROp_UIntType:   m_writer->emit("uint");   break;
-    case kIROp_Int64Type:  m_writer->emit("long");   break;
-    case kIROp_UInt64Type: m_writer->emit("ulong");  break;
-    }
+    emitSimpleTypeImpl(elementType);
 
-    if (elementCount > 1)
+    switch (elementType->getOp())
```

Comment:
I think we can get rid of this `switch` because it is not helping us achieving anything more?

Reply:
I agree to that.
I will try to remove  it next time when I need to modify this function.

Today while I was reading the document, I learned that Metal supports an expression of `vec<T,N>`. I think that will be more preferable syntax. There is also matrix<T,C,R> as well. I guess Metal is column major.

Reply:
If they are mat<T,A, B> and are A major, the we map Slang::row to A and Slang::col to B. This means that similar to Vulkan, a slang row is a metal column.

Reply:
I need to double check their definition of matrix vector multiplication and matrix constructor to make sure all these makes sense. But I expect this is likely going to be the case.